src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -59,13 +63,57 @@ get_render_value(const std::string dir)
     return instance_num;
 }
 
-
 inline pdev
 get_pcidev(const xrt_core::device* device)
 {
   return pcidev::get_dev(device->get_device_id(), device->is_userpf());
 }
 
+
+std::vector<uint64_t> 
+get_counter_status_from_sysfs(std::string mon_name_address, 
+                              std::string sysfs_file_name, 
+                              size_t size, 
+                              const xrt_core::device* device)
+{
+  auto pdev = get_pcidev(device);
+
+  /* Get full path to "name" sysfs file. 
+   * Then use that path to form full path to "counters"/"status" sysfs file 
+   * which contains counter/status data for the monitor
+   */
+  std::string name_path = pdev->get_sysfs_path(mon_name_address.c_str(), "name");
+
+  std::size_t pos = name_path.find_last_of('/');
+  std::string path = name_path.substr(0, pos+1);
+  path += sysfs_file_name;
+
+  std::vector<uint64_t> val_buf(size);
+
+  std::ifstream ifs(path);
+  if (!ifs) {
+    std::cout << "\nINFO: Incomplete counter data in " << path << std::endl;
+    return val_buf;
+  }
+
+  std::string buffer;
+  ifs >> buffer;
+
+  size_t idx = 0;
+  while (!ifs.eof()) {
+    val_buf[idx] = strtoull(buffer.c_str(), nullptr, 10);
+    idx++;
+    buffer = "";
+    ifs >> buffer;
+  }
+
+  if (idx < size)
```

Comment:
what if idx >= size ?   val_buf is sized per argument size, but what is the correlation between argument and content of sysfs node.   This is not clean I think.